hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -242,19 +286,82 @@ public void removeContainerReplica(final ContainerID id,
   @Override
   public void updateDeleteTransactionId(
       final Map<ContainerID, Long> deleteTransactionMap) throws IOException {
-    throw new UnsupportedOperationException("Not yet implemented!");
+    lock.writeLock().lock();
+    try {
+      containerStateManager.updateDeleteTransactionId(deleteTransactionMap);
+    } finally {
+      lock.writeLock().unlock();
+    }
   }
 
   @Override
   public ContainerInfo getMatchingContainer(final long size, final String owner,
-      final Pipeline pipeline, final List<ContainerID> excludedContainerIDS) {
-    throw new UnsupportedOperationException("Not yet implemented!");
+      final Pipeline pipeline, final Set<ContainerID> excludedContainerIDs) {
+    NavigableSet<ContainerID> containerIDs;
+    ContainerInfo containerInfo;
+    try {
+      synchronized (pipeline.getId()) {
```

Comment:
This is to avoid the warning that we get when we synchronize on the method parameter. This is just to fool the IDE.

In general, synchronizing on method parameter is bad. We have to fix this in a better way.